### PR TITLE
154 scheduled notifications allow users be able to add or delete the queries to use in their notification

### LINKED ISSUE #154: Scheduled Notifications: Allow users be able to add or delete the queries to use in their notification
## Is your feature request related to a problem? Please describe
In the notification page, add UI to add/delete queries 

Screenshot 2023-09-26 at 11 08 53 AM

### PR DESCRIPTION
**Pull request by jmbrunskill**
Closes #154 

# 👩🏻‍💻 What does this PR do?
 

Gives user a method to see and select sql queries to use in templates.
Any required parameters are added to the params panel to be configured.

Kapture 2023-10-11 at 16 00 23

# 🧪 How has/should this change been tested?

1.
Create SqlRecipientList (with params), Notification Query (with params)
Test different combinations of Scheduled notifications with different parameters, make sure the right combination of parameters shows in the parameter panel.

2.
Check that Coldchain still works...

## 💌 Any notes for the reviewer?

This PR still doesn't do anything with the sql queries, that will be a later PR.

### CODE REVIEW

File: frontend/packages/common/src/ui/layout/tables/DataTable.tsx [hunk 1]
```diff
@@ -108,7 +105,6 @@ export const DataTableComponent = <T extends RecordWithId>({
         display: 'flex',
         flexDirection: 'column',
         flex: 1,
-        height: '100%',
```
Comment (jmbrunskill) [inline frontend/packages/common/src/ui/layout/tables/DataTable.tsx:111]: Removing this seems a bit dangerous, but doesn't seem to have had any ill effects that I've seen so far...

**Review by mark-prins — APPROVED** [review]
🎉  Changes are working nicely - and it's all very good to use and the code looks great.

I have a couple of [layout suggestions](https://github.com/openmsupply/notify/compare/154-scheduled-notifications-allow-users-be-able-to-add-or-delete-the-queries-to-use-in-their-notification...154-scheduled-notifications-layout-suggestion)

Main thing that troubles me is that all the bits are cramped on the form.. which is unrelated to this PR really..

**Current**

Screenshot 2023-10-17 at 9 27 07 AM

**Suggested**
Screenshot 2023-10-17 at 10 16 19 AM

I've also added the end of the breadcrumb - it was showing the ID but I think it's nicer with the title

One other nit pick when using the form.. I had trouble with the validation. 
I thought that I had entered all the required fields - there was nothing else indicated in this view:

Screenshot 2023-10-17 at 9 28 21 AM

it turned out that the parameters were required - but because they aren't shown I didn't realise 🙄 

Even opening the panel - they aren't indicated as required..
Then I caused a separate problem by deleting a parameter value, but the validation allows me to save still 🤔 

Screenshot 2023-10-17 at 9 30 02 AM

File: frontend/packages/system/src/Notifications/components/NotificationQuerySelectionModal.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+  const modalHeight = Math.min(window.innerHeight - 100, 700);
+  const modalWidth = Math.min(window.innerWidth - 100, 924);
+
+  return (
+    <Modal
+      height={modalHeight}
+      width={modalWidth}
+      okButton={
+        <Tooltip title={t('label.select-queries')}>
```
[the remark this answers is not in the document]
Comment (jmbrunskill) [reply frontend/packages/system/src/Notifications/components/NotificationQuerySelectionModal.tsx:74]: I think it's a difference between `DialogButton` and `LoadingButton`
Kapture 2023-10-25 at 15 12 17

I'll change to dialog button `ok` variant like this?
image

I don't think it particularly important either way.

**Comment by jmbrunskill** [comment]
> Then I caused a separate problem by deleting a parameter value, but the validation allows me to save still 🤔

I think this one is kind of by design, and there is a comment in the code. There's a difference between a parameter with an empty string and an not filling in the field. I'm theorising that an empty string might be a valid option in some cases...
image

**Comment by jmbrunskill** [comment]
> it turned out that the parameters were required - but because they aren't shown I didn't realise 🙄
> 
> Even opening the panel - they aren't indicated as required..

image

I've put in a error state for missing parameters, doesn't automatically open the parameters tab though if you've closed it.
Interested that you've been running into that issue, is it because you are accidentally closing it, or more because you want to test things out?

I'm tempted to remove the close button as a simple fix, I also assume that only experienced people will be using the system so I think it's ok for now, and can be refined later if we get time. At least they'll get a red highlight now...

**Comment by jmbrunskill** [comment]
> 🎉 Changes are working nicely - and it's all very good to use and the code looks great.
> 
> I have a couple of [layout suggestions](https://github.com/openmsupply/notify/compare/154-scheduled-notifications-allow-users-be-able-to-add-or-delete-the-queries-to-use-in-their-notification...154-scheduled-notifications-layout-suggestion) ( popped into a suggestion branch )

Thanks heaps for the suggestions and the review!

**Pull request merged**